### ISSUE TITLE
undefined references

### ISSUE THREAD

**Issue by maxfreu** [rafaqz/Rasters.jl#321]
The vscode linter tells me that there are some undefined references in /src, a hopefully complete list:

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/lookup.jl#L196

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/openstack.jl#L26

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/polygon_ops.jl#L135

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/polygon_ops.jl#L373

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/polygon_ops.jl#L408-L411

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/methods/extract.jl#L91

https://github.com/rafaqz/Rasters.jl/blob/89b4aabfa6d7ebd5f0a298a0501b3b7320e83d06/src/methods/rasterize.jl#L252

Maybe you want to take a second look.

**Comment by rafaqz** [comment]
Thanks, yeah I need to use some limiting.Also be nice to have 100% test coverage one day...

**Comment by rafaqz** [comment]
Ok these seem to be in a few categories: obscure edge cases I never tested like reversing affine transformed dims, 3d polygons which are rare but really need testing, and actually bad bugs like `rasterize` for features. 

If you are ever interested in helping improve test coverage it would be good to do some kind of drive push this up into the high 90s at some stage. 

I also wonder if its possible to run this JET.jl in the tests so this cant happen again.

**Comment by maxfreu** [comment]
I just ran the entire package through JET, which creates a wall of text. I'm quite impressed, it not only found the bugs listed above, but also bugs in HDF5 and DiskArrays. Some of them could have been prevented by using a linter, but not all; it also traced generated code, which a linter is probably not able to do. The errors are mostly: undefined variables, "may throw xy" errors (mostly through undefined keywords / used kwargs without default), invalid builtin function calls, no method matching, trying to iterate over nothing, and some more. Maybe it's possible to filter / focus on one error type after the other, e.g. addressing undefined references first. The code I ran is super simple:

```julia
# in Rasters environment
]add Jet
using JET
report_package()
```

**Comment by rafaqz** [comment]
I've been running that too! Got quite excited by how reliable everything will be once we have it set up. But it would be good to be able to turn off "may throw" reports and other junk, because we know, those errors are intentionally there...

I also found I needed to activate the package as an environment for the dependencies to work properly.

There is also this: https://github.com/julia-vscode/StaticLint.jl

Which is I guess what you are getting in VSCode. But I havent tried it standalone.
